integrations/extensions/docs/elasticsearch-install-and-setup/python-document-ingestion/doc-ingestion.ipynb

```diff
@@ -0,0 +1,223 @@
+    "        ssl_context=context,\n",
+    "        request_timeout=480\n",
+    "    )\n",
+    "    return client\n",
+    "\n",
+    "def generate_documents(docs_dir, es_index_name, text_splitter):\n",
+    "    doc_count = 0\n",
+    "    documents = []\n",
+    "    for filename in os.listdir(docs_dir):\n",
+    "        f = os.path.join(docs_dir, filename)\n",
+    "        if os.path.isfile(f):\n",
+    "            with open(f) as file:\n",
+    "                print(\"Processing \" + f)\n",
+    "                doc_count += 1\n",
+    "                parsed_document = parser.from_file(f)\n",
+    "                document_text = parsed_document['content']\n",
+    "                chunks = text_splitter.split_text(document_text)\n",
+    "                for chunk_id, chunk_text in enumerate(chunks):\n",
+    "                    source_dict = dict()\n",
+    "                    document_dict = {\"_index\": es_index_name}\n",
+    "                    source_dict['doc_id'] = doc_count\n",
+    "                    source_dict['chunk_id'] = chunk_id\n",
+    "                    source_dict['title'] = str(filename)\n",
+    "                    source_dict['text'] = chunk_text\n",
+    "                    document_dict['_id'] = hashlib.md5(chunk_text.encode('utf-8')).hexdigest()\n",
+    "                    document_dict['_source'] = source_dict\n",
+    "                    documents.append(document_dict)\n",
+    "    return documents\n",
+    "\n",
+    "def yield_docs(documents):\n",
+    "    for doc in documents:\n",
+    "        yield doc"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "76c9b8e5-8620-4e1d-b679-892e4932f45b",
+   "metadata": {},
+   "source": [
+    "## Using LangChain text splitting for chunking"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5af516b4-e181-47d3-b9d2-bbc4f235c7fd",
+   "metadata": {},
+   "source": [
+    "We use the LangChain library to split our document text into smaller chunks. This is needed so as to fit text within our model's context window. For example, ELSER in Elasticsearch has a 512 token limit, so we need to ensure every chunk remains within that size so we can obtain embedded vectors on each of the 512 token chunks of the document text. \n",
+    "\n",
+    "You can update the code below and use different values of \"chunk_size\" and \"chunk_overlap\" depending on the downstream model that you are working with or try out other text splitters as well. You can read more about it in the [LangChain documentation](https://python.langchain.com/docs/modules/data_connection/document_transformers/)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dd94fa98-4bd0-42d0-98bb-95b6177be422",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "text_splitter =RecursiveCharacterTextSplitter.from_tiktoken_encoder(\n",
+    "    chunk_size=512,\n",
+    "    chunk_overlap=124,\n",
+    "    is_separator_regex=False,\n",
+    ")"
```

Comment:
I ran into problems with the text splitter. This is the error message:
```ValueError: Encountered text corresponding to disallowed special token '<|endoftext|>'.
If you want this text to be encoded as a special token, pass it to `allowed_special`, e.g. `allowed_special={'<|endoftext|>', ...}`.
If you want this text to be encoded as normal text, disable the check for this token by passing `disallowed_special=(enc.special_tokens_set - {'<|endoftext|>'})`.
To disable this check for all special tokens, pass `disallowed_special=()`.```

Reply:
I haven't seen this before and haven't encountered it. Can you try with a different PDF document? It seems like its encountering an `<|endoftext|>` token in the document and that seems like its coming from a generative model. What document is this?

Reply:
yeah, the document I tried ingesting is a research paper and it has this `<|endoftext|>` token. This is a rare case, but do you know if we can configure the `allowed_special` or `disallowed_special` as mentioned in the error message? 

And this is the troubled PDF: https://arxiv.org/pdf/2306.11644.pdf

Reply:
Per recommendation on this topic in the community, I added the `disallowed_special=()` since its easier as opposed to tracking multiple "allowed tokens" that users may encounter